app/controllers/bookmarks_controller.rb

```diff
@@ -0,0 +1,97 @@
+class BookmarksController < ApplicationController
+  require_login!
+  before_action :set_bookmark, only: [ :show, :edit, :update, :destroy ]
+
+  # GET /bookmarks
+  # GET /bookmarks.json
+  def index
+    @bookmarks = policy_scope(Bookmark).page params[:page]
+  end
+
+  # GET /bookmarks/new
+  def new
+    @bookmark = current_user.bookmarks.new
+    @bookmark.webpage = Webpage.new
+
+    authorize @bookmark
+  end
+
+  # POST /bookmarks
+  # POST /bookmarks.json
+  def create
+    @bookmark = current_user.bookmarks.new(bookmark_params)
+
+    authorize @bookmark
+
+    respond_to do |format|
+      if @bookmark.save
+        format.html { redirect_to @bookmark, notice: "Bookmark was successfully created." }
+        format.json { render :show, status: :created, location: @bookmark }
+      else
+        format.html { render :new }
```

Comment:
What is the semantic around http response codes for this sort of thing? Pedantically this should be an `:unprocessable_entity` too, right?

Reply:
oh I guess so

Reply:
the scaffold generator doesn't do statuses for any of the html responses, and only does the status for JSON, so I'm inclined to stick with this pattern for now